README.md
# HPAW

The HPAW project provides some projector augmented wave datasets. These datasets are mainly designed for the actinides. They were generated by using the [**Atompaw**](http://users.wfu.edu/natalie/papers/pwpaw/man.html) code. You can use them along with the [**Abinit**](https://www.abinit.org) and [**Quantum Espresso**](http://www.quantum-espresso.org) codes.

## Version

v0.1.0 (devel)

## License

GNU General Public License Version 3

## Features

Now the following elements are supported.

* Group 1
    * Fr

* Group 2
    * Ra

* Group 3
    * Lr 

* Actinides
    * Ac
    * Th
    * Pa
    * U
    * Np
    * Pu
    * Am
    * Cm
    * Bk
    * Cf
    * Es
    * Fm
    * Md
    * No

In the future, we will add datasets for more elements.

## Installation

N/A

## Documentation

See [doc/note.md](doc/note.md).

## Development

The HPAW software package is developed and maintained by the HPAW Developer Team.

Find a bug? Want to contribute? Want new features? Great! Please contact with us as soon as possible.

## Reference

N/A

## Contact

```sh
Li Huang
Institute of Materials, China Academy of Engineering Physics, Sichuan Jiangyou, PRC
email: lihuang.dmft at gmail.com
```


RELEASES.md
# HPAW

## v0.1.0 @ 7ddb105 // Jul 09, 2019

* Initialize this project


hrpaw/test/clean.sh
rm -fr w* t* l* pot* r* *.xml *.abinit expot dummy density vloc NC OCCWFN
rm -fr Fr Ra Ac Th Pa U Np Pu Am Cm Bk Cf Es Fm Md No Lr Rf


hrpaw/test/plotlog.py
import matplotlib
matplotlib.use("pdf") # setup backend
import matplotlib.pyplot as plt
import numpy as np

f, axarr = plt.subplots(2,2)
plt.subplots_adjust(hspace = 0.3, wspace = 0.3)

energy, exact, paw = np.loadtxt('logderiv.0', usecols = (0,1,2), unpack = True)
lines = axarr[0,0].plot(energy, exact, energy, paw)
plt.setp(lines[0], label = r'exact')
plt.setp(lines[1], label = r'paw')
axarr[0,0].legend()
axarr[0,0].set_title(r'$l = 0$')

energy, exact, paw = np.loadtxt('logderiv.1', usecols = (0,1,2), unpack = True)
lines = axarr[0,1].plot(energy, exact, energy, paw)
plt.setp(lines[0], label = r'exact')
plt.setp(lines[1], label = r'paw')
axarr[0,1].legend()
axarr[0,1].set_title(r'$l = 1$')

energy, exact, paw = np.loadtxt('logderiv.2', usecols = (0,1,2), unpack = True)
lines = axarr[1,0].plot(energy, exact, energy, paw)
plt.setp(lines[0], label = r'exact')
plt.setp(lines[1], label = r'paw')
axarr[1,0].legend()
axarr[1,0].set_title(r'$l = 2$')

energy, exact, paw = np.loadtxt('logderiv.3', usecols = (0,1,2), unpack = True)
lines = axarr[1,1].plot(energy, exact, energy, paw)
plt.setp(lines[0], label = r'exact')
plt.setp(lines[1], label = r'paw')
axarr[1,1].legend()
axarr[1,1].set_title(r'$l = 3$')

plt.savefig("paw_log.pdf",bbox_inches='tight')


util/plotwfn.py
#!/usr/bin/env python

##
##
## Introduction
## ============
##
## It is a python script. The purpose of this script is to visualize the
## partial-waves, PS partial-waves and projectors (wfn.i files).
##
## This script should be used by the developer only.
##
## Usage
## =====
##
## ./plotwfn.py total_number_of_projectors
##
## Author
## ======
##
## This python script is designed, created, and maintained by
##
## Li Huang // email: lihuang.dmft@gmail.com
##
## History
## =======
##
## 14/07/2019 by li huang (created)
## 14/07/2019 by li huang (last modified)
##
##

import sys
import matplotlib
matplotlib.use("pdf") # setup backend
import matplotlib.pyplot as plt
import numpy as np

# get maximum number of wave functions
argu = sys.argv[1:]
if ( len(argu) > 0 ):
    maxl = int(argu[0])
else:
    maxl = 2

# check lower bound for maxl
if maxl <= 1:
    sys.exit(-1)

# check upper bound for maxl
if maxl >= 10:
    sys.exit(-1)

# create plot layout
f, axarr = plt.subplots(3,3)
plt.subplots_adjust(hspace = 0.5, wspace = 0.3)

if maxl >= 1:
    radius, pw, ps, projector = np.loadtxt('wfn1', unpack = True)
    lines = axarr[0,0].plot(radius, pw, radius, ps, radius, projector)
    plt.setp(lines[0], label = r'$\phi_i$')
    plt.setp(lines[1], label = r'$\tilde{\phi}_i$')
    plt.setp(lines[2], label = r'$\tilde{p}_i$')
    axarr[0,0].legend()
    axarr[0,0].set_title(r'$wfn1$')

if maxl >= 2:
    radius, pw, ps, projector = np.loadtxt('wfn2', unpack = True)
    lines = axarr[0,1].plot(radius, pw, radius, ps, radius, projector)
    plt.setp(lines[0], label = r'$\phi_i$')
    plt.setp(lines[1], label = r'$\tilde{\phi}_i$')
    plt.setp(lines[2], label = r'$\tilde{p}_i$')
    axarr[0,1].legend()
    axarr[0,1].set_title(r'$wfn2$')

if maxl >= 3:
    radius, pw, ps, projector = np.loadtxt('wfn3', unpack = True)
    lines = axarr[0,2].plot(radius, pw, radius, ps, radius, projector)
    plt.setp(lines[0], label = r'$\phi_i$')
    plt.setp(lines[1], label = r'$\tilde{\phi}_i$')
    plt.setp(lines[2], label = r'$\tilde{p}_i$')
    axarr[0,2].legend()
    axarr[0,2].set_title(r'$wfn3$')

if maxl >= 4:
    radius, pw, ps, projector = np.loadtxt('wfn4', unpack = True)
    lines = axarr[1,0].plot(radius, pw, radius, ps, radius, projector)
    plt.setp(lines[0], label = r'$\phi_i$')
    plt.setp(lines[1], label = r'$\tilde{\phi}_i$')
    plt.setp(lines[2], label = r'$\tilde{p}_i$')
    axarr[1,0].legend()
    axarr[1,0].set_title(r'$wfn4$')

if maxl >= 5:
    radius, pw, ps, projector = np.loadtxt('wfn5', unpack = True)
    lines = axarr[1,1].plot(radius, pw, radius, ps, radius, projector)
    plt.setp(lines[0], label = r'$\phi_i$')
    plt.setp(lines[1], label = r'$\tilde{\phi}_i$')
    plt.setp(lines[2], label = r'$\tilde{p}_i$')
    axarr[1,1].legend()
    axarr[1,1].set_title(r'$wfn5$')

if maxl >= 6:
    radius, pw, ps, projector = np.loadtxt('wfn6', unpack = True)
    lines = axarr[1,2].plot(radius, pw, radius, ps, radius, projector)
    plt.setp(lines[0], label = r'$\phi_i$')
    plt.setp(lines[1], label = r'$\tilde{\phi}_i$')
    plt.setp(lines[2], label = r'$\tilde{p}_i$')
    axarr[1,2].legend()
    axarr[1,2].set_title(r'$wfn6$')

if maxl >= 7:
    radius, pw, ps, projector = np.loadtxt('wfn7', unpack = True)
    lines = axarr[2,0].plot(radius, pw, radius, ps, radius, projector)
    plt.setp(lines[0], label = r'$\phi_i$')
    plt.setp(lines[1], label = r'$\tilde{\phi}_i$')
    plt.setp(lines[2], label = r'$\tilde{p}_i$')
    axarr[2,0].legend()
    axarr[2,0].set_title(r'$wfn7$')

if maxl >= 8:
    radius, pw, ps, projector = np.loadtxt('wfn8', unpack = True)
    lines = axarr[2,1].plot(radius, pw, radius, ps, radius, projector)
    plt.setp(lines[0], label = r'$\phi_i$')
    plt.setp(lines[1], label = r'$\tilde{\phi}_i$')
    plt.setp(lines[2], label = r'$\tilde{p}_i$')
    axarr[2,1].legend()
    axarr[2,1].set_title(r'$wfn8$')

if maxl >= 9:
    radius, pw, ps, projector = np.loadtxt('wfn9', unpack = True)
    lines = axarr[2,2].plot(radius, pw, radius, ps, radius, projector)
    plt.setp(lines[0], label = r'$\phi_i$')
    plt.setp(lines[1], label = r'$\tilde{\phi}_i$')
    plt.setp(lines[2], label = r'$\tilde{p}_i$')
    axarr[2,2].legend()
    axarr[2,2].set_title(r'$wfn9$')

plt.savefig("paw_wfn.pdf",bbox_inches='tight')


util/plotlog.py
#!/usr/bin/env python

##
##
## Introduction
## ============
##
## It is a python script. The purpose of this script is to visualize the
## logarithmic derivatives (logderiv.l files). Note that each logderiv.l
## file correspond to l quantum number and contains the logarithmic
## derivative of the l-state, computed for exact atomic problem and with
## the PAW dataset.
##
## This script should be used by the developer only.
##
## Usage
## =====
##
## ./plotlog.py maximum_l_state
##
## Author
## ======
##
## This python script is designed, created, and maintained by
##
## Li Huang // email: lihuang.dmft@gmail.com
##
## History
## =======
##
## 14/07/2019 by li huang (created)
## 14/07/2019 by li huang (last modified)
##
##

import sys
import matplotlib
matplotlib.use("pdf") # setup backend
import matplotlib.pyplot as plt
import numpy as np

# get maximum l
argu = sys.argv[1:]
if ( len(argu) > 0 ):
    maxl = int(argu[0])
else:
    maxl = 0

# check lower bound for maximum l
if maxl <= 0:
    sys.exit(-1)

# check upper bound for maximum l
if maxl >= 4:
    sys.exit(-1)

# create plot layout
f, axarr = plt.subplots(2,2)
plt.subplots_adjust(hspace = 0.3, wspace = 0.3)

# for l == 0, s orbital
if maxl >= 0:
    energy, exact, paw = np.loadtxt('logderiv.0', usecols = (0,1,2), unpack = True)
    lines = axarr[0,0].plot(energy, exact, energy, paw)
    plt.setp(lines[0], label = r'exact')
    plt.setp(lines[1], label = r'paw')
    axarr[0,0].legend()
    axarr[0,0].set_title(r'$l = 0$')

# for l == 1, p orbital
if maxl >= 1:
    energy, exact, paw = np.loadtxt('logderiv.1', usecols = (0,1,2), unpack = True)
    lines = axarr[0,1].plot(energy, exact, energy, paw)
    plt.setp(lines[0], label = r'exact')
    plt.setp(lines[1], label = r'paw')
    axarr[0,1].legend()
    axarr[0,1].set_title(r'$l = 1$')

# for l == 2, d orbital
if maxl >= 2:
    energy, exact, paw = np.loadtxt('logderiv.2', usecols = (0,1,2), unpack = True)
    lines = axarr[1,0].plot(energy, exact, energy, paw)
    plt.setp(lines[0], label = r'exact')
    plt.setp(lines[1], label = r'paw')
    axarr[1,0].legend()
    axarr[1,0].set_title(r'$l = 2$')

# for l == 3, f orbital
if maxl >= 3:
    energy, exact, paw = np.loadtxt('logderiv.3', usecols = (0,1,2), unpack = True)
    lines = axarr[1,1].plot(energy, exact, energy, paw)
    plt.setp(lines[0], label = r'exact')
    plt.setp(lines[1], label = r'paw')
    axarr[1,1].legend()
    axarr[1,1].set_title(r'$l = 3$')

plt.savefig("paw_log.pdf",bbox_inches='tight')
